src/util/mathematical_expr.h

```diff
@@ -271,6 +271,47 @@ inline function_application_exprt &to_function_application_expr(exprt &expr)
   return ret;
 }
 
+/// \brief A base class for variable bindings (quantifiers, let, lambda)
+class binding_exprt : public binary_exprt
+{
+public:
+  using variablest = std::vector<symbol_exprt>;
+
+  /// construct the binding expression
+  binding_exprt(
+    irep_idt _id,
+    const variablest &_variables,
+    exprt _where,
+    typet _type)
+    : binary_exprt(
+        tuple_exprt((const operandst &)_variables),
+        _id,
+        std::move(_where),
+        std::move(_type))
+  {
+  }
+
+  variablest &variables()
+  {
+    return (variablest &)static_cast<tuple_exprt &>(op0()).operands();
```

Comment:
It would be possible to make this work with `USE_LIST`. Is `USE_LIST` sill useful?

Reply:
The code wouldn't actually compile with `USE_LIST`. If `USE_LIST` were to work, we could change `irept` to have only a single container and some cursor(s) into that. Whether that's more efficient or not I can't tell, because it wouldn't compile...